frontend/src/models/Question.model.tsx

```diff
@@ -5,19 +5,25 @@ export class Question {
   topics: string[];
   difficulty: number;
   id: number;
+  solved: boolean; // Field to indicate whether the question has been solved
+  solvedDate?: string;
 
   constructor(
     id: number,
     title: string,
     descMd: string,
     categories: string[],
-    difficulty: number
+    difficulty: number,
+    solved: boolean = false,
+    solvedDate?: string
   ) {
     this.title = title;
     this.descMd = descMd;
     this.topics = categories;
     this.difficulty = difficulty;
     this.id = id;
+    this.solved = solved;
+    this.solvedDate = solvedDate;
```

Comment:
instead of adding fields to our existing Question Model, would it be better to inherit from this class instead and create a new model?